src/Google.Api.Gax/ServiceSettingsBase.cs

```diff
@@ -44,6 +58,7 @@ public abstract class ServiceSettingsBase
         {
             settings.CallSettings = CallSettings.Clone();
             settings.Clock = Clock;
+            settings.UserAgent = UserAgent;
```

Comment:
Yes, that sounds like the best way to do it. The code-gen will have to change, but only slightly.